src/Components/Blazor/Blazor/src/Http/WebAssemblyHttpMessageHandler.cs

```diff
@@ -64,29 +64,22 @@ public class WebAssemblyHttpMessageHandler : HttpMessageHandler
             };
 
             options.RequestUri = request.RequestUri.ToString();
-
-            if (JSRuntime.Current is MonoWebAssemblyJSRuntime mono)
-            {
-                mono.InvokeUnmarshalled<int, byte[], string, object>(
-                    "Blazor._internal.http.sendAsync",
-                    id,
-                    request.Content == null ? null : await request.Content.ReadAsByteArrayAsync(),
-                    Json.Serialize(options));
-            }
-            else
-            {
-                throw new NotImplementedException("WebAssemblyHttpMessageHandler only supports running under Mono WebAssembly.");
-            }
+            WebAssemblyJSRuntime.Current.InvokeUnmarshalled<int, byte[], string, object>(
```

Comment:
Should we be passing this in instead of relying on a static? I can't think of a case where this would go wrong for wasm - but in general if we're telling others they can't have a static then we have to figure it out as well.

Reply:
The static is safe in the WebAssembly case, so there isn't necessarily a problem with us using it as an internal implementation detail for code paths that definitely only occur on WebAssembly.

Reply:
This specific case requires having an instance of `MonoWebAssemblyJSRuntime` not one of the abstractions such as `JSInProcessRuntimeBase` \ `IJSInProcessRuntime`. Relying on the service would mean we would have to downcast and throw \ read an optional service which just seems a bit icky. This seems a little cleaner in comparison.